scm/src/scm_input.F90

```diff
@@ -1452,11 +1465,38 @@ subroutine get_case_init_DEPHY(scm_state, scm_input)
              input_sfalb_ice       (          input_n_init_times), &
              input_emis_ice        (          input_n_init_times), &
              stat=allocate_status)
-  end if
-
+  
+  needed_for_lsm_ics = .False.
+  needed_for_model_ics = .False.
+  if (scm_state%lsm_ics .or. trim(input_surfaceForcingLSM) == "lsm") needed_for_lsm_ics = .True.
+  if (scm_state%model_ics) needed_for_model_ics = .True.
+  
   !>  - Read in the initial profiles.
-  call NetCDF_read_var(ncid, "pa", .True., input_pres)
-  call NetCDF_read_var(ncid, "zh", .True., input_height)
+  
+  if (lev_in_altitude) then
```

Comment:
@bluefinweiwei This code change is to accommodate the COMBLE case that doesn't supply everything that a DEPHY file should have. One has a choice of what to put in the 'lev' data in the NetCDF file. This code can now handle whether the data is altitude in m or pressure levels in Pa. With this change, the other changes that you had to the scm_input.F90 file are taken care of.